src/renderer/corporate/member/corporate-add-member-container.jsx

```diff
@@ -1,64 +1,187 @@
-		return this.handleSave(this.state.label, this.state.address);
-	};
+	handleFieldChange = name => evt => {
+		let value = evt;
 
-	handleSave = async (label, address) => {
-		await this.props.dispatch(addressBookOperations.addAddressBookEntry({ label, address }));
-		this.closeAction();
-	};
+		if (evt && evt.target) {
+			value = evt.target.value;
+		}
+
+		const stateErrors = { ...this.state.errors };
+		delete stateErrors[name];
+		const errors = this.validateAllAttributes([{ name, value }]);
 
-	handleLabelChange = event => {
-		event.preventDefault();
-		const label = event.target.value;
 		this.setState({
-			...this.state,
-			label
+			[name]: value
 		});
-		this.props.dispatch(addressBookOperations.validateLabel(label));
+		this.setErrors({ ...stateErrors, ...errors });
 	};
 
-	handleAddressChange = event => {
-		event.preventDefault();
-		const address = event.target.value;
+	setErrors(errors) {
+		const hasErrors = Object.keys(errors).length > 1;
 		this.setState({
-			...this.state,
-			address
+			errors: { ...errors, hasErrors }
 		});
-		this.props.dispatch(addressBookOperations.validateAddress(address));
+	}
+
+	validateAttributeType = type => {
+		return ['individual', 'corporate'].includes(type);
 	};
 
-	closeAction = () => {
-		this.props.dispatch(push('/main/create-corporate-profile'));
+	validateAttributePositions = selectedPositions => {
+		const acceptablePositions = this.props.positions.map(p => p.position);
+		let isError = false;
+		if (!selectedPositions || selectedPositions.size === 0) {
+			return false;
+		}
+		selectedPositions.forEach(p => {
+			isError = isError || !acceptablePositions.includes(p);
+		});
+		return !isError;
 	};
 
+	validateAttributeShares = shares => {
+		const number = parseInt(shares);
+		return !isNaN(number) && number >= 0 && number <= 100;
+	};
+
+	validateAttributeDid = did => true;
+
+	validateAllAttributes(attrs) {
+		const errorText = {
+			email: 'Email provided is invalid',
+			jurisdiction: 'Please select a jurisdiction',
+			entityName: 'Please enter an entity name',
+			entityType: 'Please select a entity type',
+			creationDate: 'Please enter company creation date',
+			taxId: 'Tax id provided is invalid',
+			type: 'Invalid member type',
+			positions: 'Invalid position',
+			country: 'Please select Residency',
+			nationality: 'Please select Nationality',
+			firstName: 'Please enter your first Name',
+			lastName: 'Please enter your last Name',
+			phoneNumber: 'Invalid phone number',
+			shares: 'Shares must be between 0 and 100'
+		};
+		if (!attrs) {
+			const fields = this.selectFields(this.state.type);
+			attrs = fields.map(name => ({ name, value: this.state[name] }));
+		}
+		const errors = attrs.reduce(
+			(acc, curr) => {
+				const { name, value } = curr;
+				let isError;
+
+				if (['type', 'positions', 'shares', 'did'].includes(name)) {
+					const functionName =
+						'validateAttribute' + (name.charAt(0).toUpperCase() + name.substring(1));
```

Comment:
this smells, better to do update `isValidAttribute`

if/switch/validators object by name will be good